File: src/main/java/com/freya02/botcommands/api/components/builder/PersistentComponentBuilder.java
package com.freya02.botcommands.api.components.builder;

import java.util.concurrent.TimeUnit;

public interface PersistentComponentBuilder<T extends PersistentComponentBuilder<T>> extends ComponentBuilder<T> {
	String getHandlerName();

	String[] getArgs();

	/**
	 * Makes this component expire after the specified timeout
	 * <br>Once the component expires it should be removed from the component manager
	 * <br>Additionally, components which are in the same group as this component are also deleted
	 *
	 * @return This component builder for chaining purposes
	 */
	T timeout(long timeout, TimeUnit timeoutUnit);

	PersistentComponentTimeoutInfo getTimeout();
}

File: src/main/kotlin/com/freya02/botcommands/api/new_components/builder/PersistentComponentBuilder.kt
package com.freya02.botcommands.api.new_components.builder

import com.freya02.botcommands.internal.new_components.PersistentHandler
import com.freya02.botcommands.internal.new_components.new.PersistentTimeout

//TODO use
interface PersistentComponentBuilder : ComponentBuilder {
    override val timeout: PersistentTimeout?
    override val handler: PersistentHandler?
}